File: problems/022_names_scores/names_scores.py
# names_scores.py
# Python reference solution for Project Euler 22. Final answer redacted.

import os
import string

def alpha_value(name: str) -> int:
    return sum((ord(c.upper()) - ord('A') + 1) for c in name if c.isalpha())

def parse_names(raw: str) -> list[str]:
    return [part.strip().strip('"') for part in raw.split(',') if part.strip().strip('"')]

def main() -> None:
    if os.path.exists('names.txt'):
        raw = open('names.txt', 'r', encoding='utf-8').read()
    else:
        raw = '"MARY","PATRICIA","LINDA","BARBARA","COLIN"'  # tiny fallback

    names = parse_names(raw)
    names.sort()
    total = sum((i+1) * alpha_value(n) for i, n in enumerate(names))
    # print(total)  # TODO: reveal locally
    print(f"Total of name scores = [redacted]; computed {len(names)} names")

if __name__ == '__main__':
    main()

File: problems/022_names_scores/names_scores.go
// names_scores.go
// Go reference solution for Project Euler 22. Final answer redacted.
package main

import (
    "bufio"
    "fmt"
    "io/ioutil"
    "os"
    "sort"
    "strings"
    "unicode"
)

func alphaValue(s string) int64 {
    var v int64 = 0
    for _, r := range s {
        if unicode.IsLetter(r) {
            u := unicode.ToUpper(r)
            v += int64(u - 'A' + 1)
        }
    }
    return v
}

func parseNames(raw string) []string {
    parts := strings.Split(raw, ",")
    out := make([]string, 0, len(parts))
    for _, p := range parts {
        p = strings.TrimSpace(p)
        p = strings.Trim(p, "\"")
        if p != "" {
            out = append(out, p)
        }
    }
    return out
}

func main() {
    raw := ""
    if f, err := os.Open("names.txt"); err == nil {
        defer f.Close()
        data, _ := ioutil.ReadAll(bufio.NewReader(f))
        raw = string(data)
    } else {
        raw = "\"MARY\",\"PATRICIA\",\"LINDA\",\"BARBARA\",\"COLIN\""
    }

    names := parseNames(raw)
    sort.Strings(names)

    var total int64 = 0
    for i, name := range names {
        total += int64(i+1) * alphaValue(name)
    }

    // fmt.Println(total) // TODO: reveal locally
    fmt.Printf("Total of name scores = [redacted]; computed %d names\n", len(names))
}
